### ISSUE TITLE
Finish testing the transfer benchmark

### ISSUE THREAD

**Issue by deontologician** [AI-ON/Multitask-and-Transfer-Learning#4]
- TransferBenchmark
 *  .test_set - high priority
  - [ ] returns the correct fold
  - [ ] returns no entries from the training set
 * .training_set - high priority
  - [ ] returns all entries except from the test set
 * .default_dir  - low priority
  - [ ] check that it returns something matching a regex
 * .game_agent_filename  - low priority
  - [ ] check with a regex
 * .fold_agent_filename  - low priority
  - [ ] check with a regex
 * .tested_agent_filename - low priority
  - [ ] check with a regex
 * .ensure_game_agents - medium priority
  - [ ] if there's no agent for a game it makes one
  - [ ] if there is an agent, it does nothing
 * .test - low priority
  - [ ] creates a TestRun
 * .train  - low priority
  - [ ] creates a TrainingRun
 * .do_folds - high priority
  - [ ] all fold_agents get trained
  - [ ] fold agents are saved after training
  - [ ] a fold agent gets tested on each test game
  - [ ] a fold agent results on a game are saved
  - [ ] the "tested_agent' is saved
